Kontent.Ai.Delivery.Extensions.DependencyInjection/Builders/MultipleDeliveryClientFactoryBuilder.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Concurrent;
+using Kontent.Ai.Delivery.Abstractions;
+using Kontent.Ai.Delivery.Builders;
+using Kontent.Ai.Delivery.Builders.DeliveryClient;
+using Kontent.Ai.Delivery.Caching;
+using Kontent.Ai.Delivery.Configuration;
+
+namespace Kontent.Ai.Delivery.Extensions.DependencyInjection.Builders
+{
+    internal class MultipleDeliveryClientFactoryBuilder : IMultipleDeliveryClientFactoryBuilder
+    {
+        private readonly ConcurrentDictionary<string, IDeliveryClient> _clients = new();
+
+        public IMultipleDeliveryClientFactoryBuilder AddDeliveryClient
+        (
+            string name,
+            Func<IDeliveryOptionsBuilder,
+            DeliveryOptions> deliveryOptionsBuilder,
+            Func<IOptionalClientSetup, IOptionalClientSetup> optionalClientSetup = null
+        )
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (deliveryOptionsBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(deliveryOptionsBuilder));
+            }
+
+            var client = BuildDeliveryClient(deliveryOptionsBuilder, optionalClientSetup);
+            RegisterClient(name, client);
+
+            return this;
+        }
+
+        public IMultipleDeliveryClientFactoryBuilder AddDeliveryClientCache
+        (
+            string name, Func<IDeliveryOptionsBuilder,

```

Comment:
One parameter per line would be better.

Reply:
6cdbcc6

Reply:
There are 4 parameters and 4 lines but still, it is not one parameter per line. 

```
            string name,
            Func<IDeliveryOptionsBuilder, DeliveryOptions> deliveryOptionsBuilder,
            IDeliveryCacheManager cacheManager,
            Func<IOptionalClientSetup, IOptionalClientSetup> optionalClientSetup = null
```